azurerm/internal/services/datafactory/data_factory_linked_service_azure_sql_database_resource.go

```diff
@@ -0,0 +1,280 @@
+				ValidateFunc:     validation.StringIsNotEmpty,
+			},
+
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"integration_runtime_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"service_principal": {
+				Type: schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type: schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+						"password": {
+							Type: schema.TypeString,
+							Required: true,
+							Sensitive: true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+						"tenant": {
+							Type: schema.TypeString,
+							
+						},
+					},
+				},
+			},
+
+			"parameters": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"annotations": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"additional_properties": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func resourceArmDataFactoryLinkedServiceAzureSQLDatabaseCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DataFactory.LinkedServiceClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	dataFactoryName := d.Get("data_factory_name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, dataFactoryName, name, "")
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("checking for presence of existing Data Factory Linked Service AzureSQLDatabase %q (Data Factory %q / Resource Group %q): %+v", name, dataFactoryName, resourceGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_data_factory_linked_service_azure_sql_database", *existing.ID)
+		}
+	}
+
+	connectionString := d.Get("connection_string").(string)
+	secureString := datafactory.SecureString{
+		Value: &connectionString,
+		Type:  datafactory.TypeSecureString,
+	}
+
+	sqlProperties := &datafactory.AzureSQLDatabaseLinkedServiceTypeProperties{
+		ConnectionString: &secureString,
+	}
+
+	description := d.Get("description").(string)
+
+	azureSQLDatabaseLinkedService := &datafactory.AzureSQLDatabaseLinkedService{
+		Description: &description,
+		AzureSQLDatabaseLinkedServiceTypeProperties: sqlProperties,
+		Type: datafactory.TypeAzureSQLDatabase,
+	}
```

Comment:
minor and optional i'd prefer this to be
```suggestion
	azureSQLDatabaseLinkedService := &datafactory.AzureSQLDatabaseLinkedService{
		Description: &util.string(d.Get("description").(string)),,
		AzureSQLDatabaseLinkedServiceTypeProperties: &datafactory.AzureSQLDatabaseLinkedServiceTypeProperties{
			ConnectionString: &datafactory.SecureString{
				Value: util.string(d.Get("connection_string").(string)),
				Type:  datafactory.TypeSecureString,
			},
		}
		Type: datafactory.TypeAzureSQLDatabase,
	}
```